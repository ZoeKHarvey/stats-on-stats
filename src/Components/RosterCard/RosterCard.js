import './RosterCard.scss'
import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux'


 export const RosterCard = ({  jerseyNumber, person, position, getSinglePlayer }) => {
  
	return (
		<section className='roster-card'>
    <p className="roster__p--name">{person.fullName}</p>
    <p className="roster__p--number">{jerseyNumber}</p>
    <p className="roster__p--pname">{position.name}</p>
    <Link to='/player'>
    <button className="player__button" onClick={(e) => getSinglePlayer(e, person.id)}>ShowPlayer</button>
    </Link>
		</section>

	)
}

export const mapStateToProps = (state) => ({
  teams:state.teams,
  // teamSchedule: state.teamSchedule
})

export default connect(mapStateToProps)(RosterCard);