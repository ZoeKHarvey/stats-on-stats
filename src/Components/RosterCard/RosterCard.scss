.roster__div--schedule {
  display: flex;
}

.roster-card {
  &:hover {
    background-color: white;
  }
  display: flex;
    flex-direction: column;
    width: 11%;
    text-align: center;
    border: 1px solid;
    border-radius: 35px;
    font-family: "Barlow Condensed", sans-serif;
    margin-left: 1%;
}

.roster__p--number {
  font-size: 2em;
  margin-bottom: 3%;
  margin-top: 0;
}

.roster__p--name {
  font-weight: bold;
  font-size: 1.3em;
  margin-bottom: 0;
}

.roster__p--pname {
  margin-top: 0;
  margin-bottom: 3%;
}

.player__button {
  &:hover {
    transform: scale(1.1);
    font-weight: bold;
    border: 1px solid black;
  }
  height: 3vh;
  border-radius: 17px;
  font-family: 'Barlow Condensed', sans-serif;
}