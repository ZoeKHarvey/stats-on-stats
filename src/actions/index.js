export const getTeams = (teams) => ({
  type:'GET_TEAMS',
  teams
})

export const getStandings = (standings) => ({
  type: 'GET_STANDINGS',
  standings
})

export const getRoster = (roster) => ({
  type: 'GET_ROSTER',
  roster
})

export const getPlayer = (player) => ({
  type: 'GET_PLAYER',
  player
})

export const getPlayerStats = (player) => ({
  type: 'GET_PLAYER_STATS',
  player
})

export const getTeamSchedule = (schedule) => ({
  type: 'GET_TEAM_SCHEDULE',
  schedule
})

export const getPlayerProjections = (player) => ({
  type: 'GET_PLAYER_PROJECTIONS',
  player
})

export const favoriteTeams = (newTeam) => ({
  type: 'GET_FAVORITE_TEAMS',
  newTeam
})

export const isLoading = (isLoading) => ({
  type: 'IS_LOADING',
  isLoading
})

export const hasError = (errorMsg) => ({
  type: 'HAS_ERROR',
  errorMsg
})



