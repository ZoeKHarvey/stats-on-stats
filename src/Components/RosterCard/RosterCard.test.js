import React from 'react';
import { shallow } from 'enzyme';
import { RosterCard } from './RosterCard';
import { getPlayer } from '../../actions'

describe('RosterCard', () => {
  let wrapper;

  let person = {
    fullName: 'Vito Burrito',
  }

  let position = {
    type: 'Defense'
  }

  beforeEach(() => {
    wrapper = shallow (
      <RosterCard person={person}
                  position={position} />
    )
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
})


