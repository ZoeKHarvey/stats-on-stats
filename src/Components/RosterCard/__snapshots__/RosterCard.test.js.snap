// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RosterCard should match the snapshot 1`] = `ShallowWrapper {}`;
